### PR TITLE
ANDROID-10988 Change 'Link checker' trigger on PR instead of dispatch and fix bad links

### PR DESCRIPTION
**Pull request by haynlo**
### :goal_net: What's the goal?
Trigger 'Link checker' workflow validation on Pull Request and fix bad links.

### :construction: How do we do it?
Change logic in .yml file
Fix bad links reported by workflow

### :test_tube: How can I test this?
link_checker

### CODE REVIEW

File: library/src/main/java/com/telefonica/mistica/compose/README.md [hunk 1]
```diff
@@ -12,27 +12,27 @@ repositories {
...
-    ...
+    // ...
 }
 ```
 
 ## Configuration
 
 The goal of the team is to provide a fully compose implementation of the library, but currently there are still wrapped views inside `AndroidView` composables.
-So it is still needed to add [the brand theme in the Manifest](../README.md#configuration)
+So it is still needed to add [the brand theme in the Manifest](../../../../../../../../README.md#configuration)
```
Comment (dagonco) [inline library/src/main/java/com/telefonica/mistica/compose/README.md:35]: Probably [...](#configuration) it's enough to reference configuration section.

**Review by dagonco — APPROVED** [review]
Great job! 🚀

File: library/src/main/java/com/telefonica/mistica/compose/README.md (hunk 1 above)
Comment (haynlo) [reply library/src/main/java/com/telefonica/mistica/compose/README.md:35]: I made a quick test and it didn't work as expected :(

If I declare only `(#configuration)` it only points to the `#configuration` above in the same file, and I want to point to the `#configuration` from the top level `README.md`

File: .github/workflows/link_checker.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@ jobs:
 
       - name: Link Checker
         id: lychee
-        uses: lycheeverse/lychee-action@v1.5.1
+        uses: lycheeverse/lychee-action@v1.6.1
```
Comment (jeslat) [inline .github/workflows/link_checker.yml:14]: I would put just the major version number:
```
uses: lycheeverse/lychee-action@v1
```

File: README.md [hunk 3]
```diff
@@ -22,20 +22,20 @@ repositories {
     mavenCentral()
 }
 dependencies {
-    ...
+    // ...
```
Comment (jeslat) [inline README.md:25]: If the three points are generating problems I think we can remove them. In my opinion, just the `implementation` line is enough to understand how to add the dependency.

File: library/src/main/java/com/telefonica/mistica/compose/README.md [hunk 4]
```diff
@@ -12,27 +12,27 @@ repositories {
     mavenCentral()
 }
 dependencies {
-    ...
+    // ...
```
Comment (jeslat) [inline library/src/main/java/com/telefonica/mistica/compose/README.md:15]: Same here, I would remove the three points

**Review by jeslat — APPROVED** [review]
Good job!

**Pull request merged**